phylogeny/tests/test_prototypeSelection.py

```diff
@@ -0,0 +1,139 @@
+from unittest import TestCase, main
+from phylogeny.prototypeSelection import (prototypeSelection_exhaustive,
+                                          distanceSum)
+from skbio.stats.distance import DistanceMatrix
+from skbio.stats.distance._base import (DistanceMatrixError,
+                                        DissimilarityMatrixError,
+                                        MissingIDError)
+from skbio.util import get_data_path
+from skbio.io._exception import UnrecognizedFormatError
+
+
+class prototypeSelection(TestCase):
+    def setUp(self):
+        self.dm_nonNull = get_data_path('distMatrix_nonNull.txt')
+        self.dm_repIDs = get_data_path('distMatrix_repIDs.txt')
+        self.dm_asym = get_data_path('distMatrix_asym.txt')
+        # this file must not exists!
+        self.dm_noFile = get_data_path('noFile.txt')
+        # any file that is present, but not a DistanceMatrix
+        self.dm_wrongFormat = get_data_path('wrongFileformat.txt')
+
+        self.dm100 = DistanceMatrix.read(get_data_path('distMatrix_100.txt'))
+        self.dm20 = DistanceMatrix.read(get_data_path('distMatrix_20_f5.txt'))
+
+    def test_distanceSum(self):
+        # test that no missing IDs can be used
+        self.assertRaisesRegex(
+            MissingIDError,
+            'The ID \'X\' is not in the dissimilarity matrix.',
+            distanceSum,
+            ['A', 'B', 'X'],
+            self.dm20)
+
+        # test that no ID is duplicated
+        self.assertRaisesRegex(
+            Exception,
+            'List of elements contain duplicates!',
+            distanceSum,
+            ['A', 'B', 'C', 'D', 'B'],
+            self.dm20)
+
+        # test that list of IDs holds at least 2 elements
+        self.assertRaises(
+            AssertionError,
+            distanceSum,
+            ['A'],
+            self.dm20)
+
+        # test that list of IDs holds at most as many elements as the distance
+        # matrix
+        self.assertRaises(
+            AssertionError,
+            distanceSum,
+            range(0, 21),
+            self.dm20)
+
+        # test for correct type
+        self.assertRaises(
+            TypeError,
+            distanceSum,
+            range(0, 21),
+            None)
+
+    def test_exhaustive(self):
+        # check if execution is rejected if number of combination is too high
+        self.assertRaisesRegex(
+            Exception,
+            'Cowardly refuse to test ',
+            prototypeSelection_exhaustive,
+            self.dm20,
+            5,
+            maxCombinationsToTest=1000)
```

Comment:
implicitly by the testing correctness of the return value of the function in the following tests.